src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ProblemMessageControllerTest.java

```diff
@@ -0,0 +1,205 @@
+    private LaunchDarklyFeatureFlagProvider launchDarklyFeatureFlagProvider;
+
+    @BeforeEach
+    void setUp() {
+        when(launchDarklyFeatureFlagProvider.getBooleanValue(any(), any())).thenReturn(true);
+        when(authTokenGenerator.generate()).thenReturn(S2S_TOKEN);
+    }
+
+    @NotNull
+    private String randomMessageId() {
+        return "messageId_" + ThreadLocalRandom.current().nextLong(1000000);
+    }
+
+    @Test
+    void should_return_a_ready_message_when_ready_message_timestamp_is_old_than_one_hour() throws Exception {
+
+        String messageId = randomMessageId();
+
+        when(launchDarklyFeatureFlagProvider.getBooleanValue(any(), any())).thenReturn(true).thenReturn(true);
+
+        when(deadLetterQueuePeekService.isDeadLetterQueueEmpty()).thenReturn(true);
+
+        postAnOldMessage(messageId, status().isCreated(), false);
+
+        await().pollInterval(600, MILLISECONDS)
+            .atMost(45, SECONDS)
+            .untilAsserted(() -> checkMessagesInState(List.of(messageId), MessageState.READY));
+
+        MvcResult mvcResult = postAJobRequest(JobName.FIND_PROBLEM_MESSAGES.name());
+        List<CaseEventMessage> response = OBJECT_MAPPER.readValue(mvcResult.getResponse().getContentAsString(),
+                                                        new TypeReference<>() {});
+
+        List<MessageState> messageStates = response.stream().map(caseEventMessage1 -> caseEventMessage1.getState())
+            .collect(Collectors.toList());
+        assertNotNull(mvcResult, "Response should not be null");
+        assertTrue(messageStates.contains(MessageState.READY));
+    }
+
+    @Test
+    void should_return_unprocessable_message_when_case_id_is_null() throws Exception {
+        String messageId = randomMessageId();
+        String unprocessableMessage = getCaseEventMessage(null);
+
+        MvcResult result = mockMvc.perform(post("/messages/" + messageId + "?from_dlq=true")
+                                               .contentType(MediaType.APPLICATION_JSON)
+                                               .content(unprocessableMessage))
+            .andExpect(status().isCreated())
+            .andReturn();
+
+
+        String content = result.getResponse().getContentAsString();
+        assertEquals(201, result.getResponse().getStatus(), content);
+        assertNotNull(content, "Content Should not be null");
+
+        MvcResult mvcResult = postAJobRequest(JobName.FIND_PROBLEM_MESSAGES.name());
+        List<CaseEventMessage> response = OBJECT_MAPPER.readValue(mvcResult.getResponse().getContentAsString(),
+                                                        new TypeReference<>() {});
+        List<MessageState> messageStates = response.stream().map(caseEventMessage1 -> caseEventMessage1.getState())
+            .collect(Collectors.toList());
+
+        assertNotNull(mvcResult, "Response should not be null");
+        assertTrue(messageStates.contains(MessageState.UNPROCESSABLE));
+    }
+
```

Comment:
Can we add 2 tests to cover the following;
1. When there no unprocessable messages found.
2. When the are no messages where ready state == ready & > 1hr.

This will firm up that our native query in the repository filters correctly based on scenarios.

Reply:
Added more tests to cover these scenarios.